File: 1164.py
n = int(input())
a = 0
lista=[]
numeros = []
for i in range(0, n):
    num = int(input())
    numeros+=[num]
    for j in range(1, num):
        if num%j==0:
            a+=j
    if a == num:
        lista+=[True]
    else:
        lista+=[False]
    a = 0
for l in range(0, n):
    print(f'{numeros[l]} eh perfeito') if lista[l] == True else print(f'{numeros[l]} nao eh perfeito')

File: 1164.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split('\n');

let n = Number(lines.shift());
let a = 0;
let lista = [];
let numeros = [];
for(var i = 0; i < n; i++){
    let num = Number(lines.shift());
    numeros.push(num);
    for(var j = 1; j < num; j++){
        if (num % j === 0){
            a += j;
        }
    }
    if(a === num){
        lista.push(true);
    }
    else{
        lista.push(false);
    }
    a = 0;
}
for(var l = 0; l < n; l++){
    lista[l] === true ? console.log(`${numeros[l]} eh perfeito`): console.log(`${numeros[l]} nao eh perfeito`);
}
